docs/ref/applications.txt

```diff
@@ -509,3 +514,26 @@ Here are some common problems that you may encounter during initialization:
   :setting:`INSTALLED_APPS` to contain
   ``'django.contrib.admin.apps.SimpleAdminConfig'`` instead of
   ``'django.contrib.admin'``.
+
+* ``RuntimeWarning: Accessing the database during app initialization is
+  discouraged.`` This happens if one of your apps attempts database access in
+  either the :meth:`AppConfig.ready` method or during import time. This is bad
+  because it will happen during the startup of every management command.
```

Comment:
Rewording + adding more reasons:

```suggestion
* ``RuntimeWarning: Accessing the database during app initialization is
  discouraged.`` This warning is triggered for database queries that run before
  apps are ready, such as during module imports. Such premature database
  queries are discouraged because they will slow down your project startup,
  potentially cache stale data, and can even fail if migrations are pending.
```

I even helped a client last week who had an import-time database query that started failing after they added a migration. All their processes failed, including `./manage.py migrate`, so they couldn't actually apply the migration without a workaround!

Reply:
I kept the reference to `AppConfig.ready()` and the part that the queries run at the startup of every management command if that's okay.

Reply:
Y“During the startup of every management command“ is true, but slightly less than complete, because processes that load the project for you, like gunicorn and pytest, aren’t management commands but are still affected. This is why I suggested the generic “your project startup“.

I’m happy to leave as “every management command“, but perhaps there’s other wording like “every process“ or “every command“? I'm not sure what's clearest.

Reply:
Since the warning block below `AppConfig.ready()` also talks about management commands, I would also leave it as "management commands" here. Unless you think we should rewrite the warning block as well.

https://github.com/django/django/blob/f06eafa94744229b1e44c2029aa3a1f4cc826adf/docs/ref/applications.txt#L301-L312

Reply:
Yeah okay, let's stick with it.